pycaching/cache.py

```diff
@@ -853,7 +878,12 @@ def load_quick(self):
         # prettify data
         self.name = data["name"]
         self.type = Type.from_string(data["type"]["text"])
-        self.state = data["available"]
+
+        # We can fill in status correctly only for enabled caches
```

Comment:
Did you add a test for this as well?

Reply:
Now I'm trying and I ran into this issue:

<details>
    <summary>Stack trace</summary>

```
____________________________________________ TestCacheStatus.test_cache_status ____________________________________________

self = <pycaching.geocaching.Geocaching object at 0x10dd52eb0>, url = 'http://tiles01.geocaching.com/map.details'
expect = 'json', method = 'GET', login_check = True, kwargs = {'params': {'i': 'GC8CKQQ'}}, res = <Response [500]>

    def _request(self, url, *, expect="soup", method="GET", login_check=True, **kwargs):
        """
        Do a HTTP request and return a response based on expect param.
    
        :param str url: Request target.
        :param str method: HTTP method to use.
        :param str expect: Expected type of data (either :code:`soup`, :code:`json` or :code:`raw`).
        :param bool login_check: Whether to check if user is logged in or not.
        :param kwargs: Passed to `requests.request
            <http://docs.python-requests.org/en/latest/api/#requests.request>`_ as is.
        """
        # check login unless explicitly turned off
        if login_check and not self._logged_in:
            raise NotLoggedInException("Login is needed.")
    
        url = url if "//" in url else urljoin(self._baseurl, url)
    
        try:
            res = self._session.request(method, url, **kwargs)
>           res.raise_for_status()

pycaching/geocaching.py:83: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <Response [500]>

    def raise_for_status(self):
        """Raises :class:`HTTPError`, if one occurred."""
    
        http_error_msg = ""
        if isinstance(self.reason, bytes):
            # We attempt to decode utf-8 first because some servers
            # choose to localize their reason strings. If the string
            # isn't utf-8, we fall back to iso-8859-1 for all other
            # encodings. (See PR #3538)
            try:
                reason = self.reason.decode("utf-8")
            except UnicodeDecodeError:
                reason = self.reason.decode("iso-8859-1")
        else:
            reason = self.reason
    
        if 400 <= self.status_code < 500:
            http_error_msg = (
                f"{self.status_code} Client Error: {reason} for url: {self.url}"
            )
    
        elif 500 <= self.status_code < 600:
            http_error_msg = (
                f"{self.status_code} Server Error: {reason} for url: {self.url}"
            )
    
        if http_error_msg:
>           raise HTTPError(http_error_msg, response=self)
E           requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: https://tiles01.geocaching.com/map.details?i=GC8CKQQ

.venv/lib/python3.8/site-packages/requests/models.py:1021: HTTPError

The above exception was the direct cause of the following exception:

self = <test.test_cache.TestCacheStatus testMethod=test_cache_status>

    def test_cache_status(self):
        with self.subTest("Enabled"):
            cache = Cache(self.gc, "GC8CKQQ")
            with self.recorder.use_cassette("cache_status_enabled"):
                self.assertEqual(Status.enabled, cache.status)
    
        with self.subTest("Disabled"):
            cache = Cache(self.gc, "GC7Y77T")
            with self.recorder.use_cassette("cache_status_disabled"):
                self.assertEqual(Status.disabled, cache.status)
    
        with self.subTest("Archived"):
            cache = Cache(self.gc, "GC1PAR2")
            with self.recorder.use_cassette("cache_status_archived"):
                self.assertEqual(Status.archived, cache.status)
    
        with self.subTest("Locked"):
            cache = Cache(self.gc, "GC10")
            with self.recorder.use_cassette("cache_status_locked"):
                self.assertEqual(Status.locked, cache.status)
    
        with self.subTest("Enabled > Quick load"):
            cache = self.gc.get_cache("GC8CKQQ")
>           cache.load_quick()

test/test_cache.py:498: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pycaching/cache.py:860: in load_quick
    res = self.geocaching._request(self._urls["tiles_server"], params={"i": self.wp}, expect="json")
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pycaching.geocaching.Geocaching object at 0x10dd52eb0>, url = 'http://tiles01.geocaching.com/map.details'
expect = 'json', method = 'GET', login_check = True, kwargs = {'params': {'i': 'GC8CKQQ'}}, res = <Response [500]>

    def _request(self, url, *, expect="soup", method="GET", login_check=True, **kwargs):
        """
        Do a HTTP request and return a response based on expect param.
    
        :param str url: Request target.
        :param str method: HTTP method to use.
        :param str expect: Expected type of data (either :code:`soup`, :code:`json` or :code:`raw`).
        :param bool login_check: Whether to check if user is logged in or not.
        :param kwargs: Passed to `requests.request
            <http://docs.python-requests.org/en/latest/api/#requests.request>`_ as is.
        """
        # check login unless explicitly turned off
        if login_check and not self._logged_in:
            raise NotLoggedInException("Login is needed.")
    
        url = url if "//" in url else urljoin(self._baseurl, url)
    
        try:
            res = self._session.request(method, url, **kwargs)
            res.raise_for_status()
    
            # return bs4.BeautifulSoup, JSON dict or raw requests.Response
            if expect == "soup":
                return bs4.BeautifulSoup(res.text, "html.parser")
            elif expect == "json":
                return res.json()
            elif expect == "raw":
                return res
    
        except requests.exceptions.RequestException as e:
            if e.response.status_code == 429:  # Handle rate limiting errors
                raise TooManyRequestsError(
                    url, rate_limit_reset=int(e.response.headers.get("x-rate-limit-reset", "0"))
                ) from e
    
>           raise Error("Cannot load page: {}".format(url)) from e
E           pycaching.errors.Error: Cannot load page: http://tiles01.geocaching.com/map.details

pycaching/geocaching.py:99: Error
```

</details>

The test code is following:
```py
        with self.subTest("Enabled > Quick load"):
            cache = Cache(self.gc, "GC8CKQQ")
            cache.load_quick()
            with self.recorder.use_cassette("cache_status_enabled_load_quick"):
                self.assertEqual(Status.enabled, cache.status)
```

---

But this works without any problems:

```py
import pycaching
from pycaching import Cache

geocaching = pycaching.login("USERNAME", "PASSWORD")

cache = Cache(geocaching, "GC8CKQQ")
cache.load_quick()
print(cache.status)
```

I can also load the [URL](https://tiles01.geocaching.com/map.details?i=GC8CKQQ) in the browser and it shows the needed JSON.

Reply:
There seems to be some issue with the cookies sent by the application:

```
[HttpRequestValidationException]: A potentially dangerous Request.Cookies value was detected from the client (gspkauth=&quot;&lt;AUTH COOKIE&gt;&quot;)
```

Nevertheless, I am not sure why this is the case here. I am going to open a corresponding issue.

For the time being, just uncomment the remaining code of the corresponding test class and record the cassette, then remove the comment markers again. Additionally, your code should call `cache.load_quick()` inside the recorder context instead of doing it beforehand, as the cassette would be empty otherwise.

Reply:
Thanks for the help :)
I added it in https://github.com/tomasbedrich/pycaching/pull/193/commits/cd51a3e84245c4951ab943de2c486d7250f6c51c